pkg/requester/ranking/utils.go

```diff
@@ -0,0 +1,19 @@
+package ranking
+
+import (
+	"context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/requester"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+)
+
+const logLevel = zerolog.TraceLevel
+
+func logRank(ctx context.Context, rank requester.NodeRank) {
```

Comment:
nit: I'm not keen on have a separate function specifically for logging. Could we instead implement [`LogObjectMarshaler`](https://github.com/rs/zerolog/blob/9070d49a1a4154c819e9c5ae6c923f3cc916d959/event.go#L48) on the `requester.NodeRank` type and then log it without this method like:
```go
log.Ctx(ctx).Trace().Object(rank).Msg("Ranked node")
```

We could also use `LogArrayMarshaler` if we are logging an array of these structures.

Reply:
Cool, did'nt know you could do that!